airflow/ui/src/queries/useLogs.tsx

```diff
@@ -87,6 +91,16 @@ const renderStructuredLog = ({
     return "";
   }
 
+  elements.push(
+    <Link
+      key={`line_${index}`}
+      to={`/dags/${taskInstance?.dag_id}/runs/${taskInstance?.dag_run_id}/tasks/${taskInstance?.task_id}#${index}`}
```

Comment:
This is a problem withthe generated URL for mapped tasks - need to add the map-index as well

Reply:
Thanks, switched to `getTaskInstanceLink` from `src/utils/links.ts` which should handle this.